examples/seismic/tutorials/05_dask.ipynb

```diff
@@ -0,0 +1,496 @@
+    "\n",
+    "\n",
+    "from examples.seismic.acoustic import AcousticWaveSolver\n",
+    "from examples.seismic import RickerSource, Receiver\n",
+    "\n",
+    "# This is used by the worker to get the current model. \n",
+    "def get_current_model(param):\n",
+    "    \"\"\" Returns the current model.\n",
+    "    \"\"\"\n",
+    "    model = get_initial_model()\n",
+    "    model.m.data[:] = np.reshape(np.load(param['model']), model.m.data.shape)\n",
+    "    return model\n",
+    "\n",
+    "# Inversion crime alert! Here the worker is creating the 'observed' data\n",
+    "# using the real model. For a real case the worker would be reading\n",
+    "# seismic data from disk.\n",
+    "def get_data(param):\n",
+    "    \"\"\" Returns source and receiver data for a single shot labeled 'shot_id'.\n",
+    "    \"\"\"\n",
+    "    true_model = get_true_model()\n",
+    "    dt = true_model.critical_dt  # Time step from model grid spacing\n",
+    "\n",
+    "    # Set up source data and geometry.\n",
+    "    nt = int(1 + (param['tn']-param['t0']) / dt)  # Discrete time axis length\n",
+    "\n",
+    "    src = RickerSource(name='src', grid=true_model.grid, f0=param['f0'],\n",
+    "                       time=np.linspace(param['t0'], param['tn'], nt))\n",
+    "    src.coordinates.data[0, :] = [30, param['shot_id']*1000./(param['nshots']-1)]\n",
+    "\n",
+    "    # Set up receiver data and geometry.\n",
+    "    nreceivers = 101  # Number of receiver locations per shot\n",
+    "    rec = Receiver(name='rec', grid=true_model.grid, npoint=nreceivers, ntime=nt)\n",
+    "    rec.coordinates.data[:, 1] = np.linspace(0, true_model.domain_size[0], num=nreceivers)\n",
+    "    rec.coordinates.data[:, 0] = 980. # 20m from the right end\n",
+    "\n",
+    "    # Set up solver - using model_in so that we have the same dt,\n",
+    "    # otherwise we should use pandas to resample the time series data. \n",
+    "    solver = AcousticWaveSolver(true_model, src, rec, space_order=4)\n",
+    "\n",
+    "    # Generate synthetic receiver data from true model\n",
+    "    true_d, _, _ = solver.forward(src=src, m=true_model.m)\n",
+    "\n",
+    "    return src, true_d, nt, solver"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dask specifics\n",
+    "\n",
+    "Previously we defined a function to calculate the individual contribution to the functional and gradient for each shot, which was then used in a loop over all shots. However, when using distributed frameworks such as Dask we instead think in terms of creating a worklist which gets *mapped* onto the worker pool. The sum reduction is also performed in parallel. For now however we assume that the scipy.optimize.minimize itself is running on the *master* process; this is a reasonable simplification because of computational cost of calculating (f, g) far exceeds off of the other compute costs."
```

Comment:
"because of computational cost of calculating (f, g) far exceeds off of the other compute costs." -> because the computational cost of calculating (f, g) far exceeds the other compute costs

Reply:
done